src/luarocks/build/builtin.lua

```diff
@@ -218,20 +256,9 @@ function builtin.run(rockspec)
          end
       end
       if type(info) == "table" then
-         local objects = {}
-         local sources = info.sources
-         if info[1] then sources = info end
-         if type(sources) == "string" then sources = {sources} end
-         for _, source in ipairs(sources) do
-            local object = source:gsub("%.[^.]*$", "."..cfg.obj_extension)
-            if not object then
-               object = source.."."..cfg.obj_extension
-            end
-            ok = compile_object(object, source, info.defines, info.incdirs)
-            if not ok then
-               return nil, "Failed compiling object "..object
-            end
-            table.insert(objects, object)
+         local objects = build_objects(info, compile.dynamic_object)
```

Comment:
This section should be conditional on `if flags["dynamic"] then`

Reply:
shouldn't I add `or flags["dynamic"] and flags["static"]` to get dynamic libs always?

Reply:
Do you always want dynamic libs?

If I write `luarocks make --static` I don't really expect a dynamic lib to come out as well.
That's that `luarocks make --static --dynamic` would be for.